core/src/main/scala/org/apache/spark/scheduler/TaskSetBlacklist.scala

```diff
@@ -61,6 +61,16 @@ private[scheduler] class TaskSetBlacklist(val conf: SparkConf, val stageId: Int,
   private val blacklistedExecs = new HashSet[String]()
   private val blacklistedNodes = new HashSet[String]()
 
+  private var latestFailureReason: String = null
+
+  /**
+   * Get the most recent failure reason of this TaskSet.
+   * @return
+   */
+  def getLatestFailureReason: String = {
```

Comment:
@jerryshao the whole class is `private[scheduler]`, so I think is OK.

Reply:
@squito yes from scope level it is fine. My thought is that this exposes the class member to other class unnecessarily. Yeah it is not a big deal, just my personal preference.

Reply:
Thanks @jerryshao @squito Could you help trigger an other jenkins test?Since last one has pySpark failure.